python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py

```diff
@@ -0,0 +1,330 @@
+    The second convenience attribute is named remove. You may assign an
+    iterable of strings which correspond to attribute names on the
+    `ConfigurableActionStruct`. All of the corresponding attributes will then
+    be removed. If any attribute does not exist, an `AttributeError` will be
+    raised. Any attributes in the Iterable prior to the name which raises will
+    have been removed from the `ConfigurableActionStruct`
+    """
+    # declare attributes that are set with __setattr__
+    _config: Config
+    _attrs: Dict[str, ConfigurableAction]
+    _field: ConfigurableActionStructField
+    _history: List[tuple]
+
+    # create descriptors to handle special update and remove behavior
+    update = ConfigurableActionStructUpdater()
+    remove = ConfigurableActionStructRemover()
+
+    def __init__(self, config: Config, field: ConfigurableActionStructField,
+                 value: Mapping[str, ConfigurableAction], at: Any, label: str):
+        object.__setattr__(self, '_config', config)
+        object.__setattr__(self, '_attrs', {})
+        object.__setattr__(self, '_field', field)
+        object.__setattr__(self, '_history', [])
+
+        self.history.append(("Struct initialized", at, label))
+
+        if value is not None:
+            for k, v in value.items():
+                setattr(self, k, v)
+
+    @property
+    def history(self) -> List[tuple]:
+        return self._history
+
+    @property
+    def fieldNames(self) -> Iterable[str]:
+        return self._attrs.keys()
+
+    def __setattr__(self, attr: str, value: Union[ConfigurableAction, Type[ConfigurableAction]],
+                    at=None, label='setattr', setHistory=False) -> None:
+
+        if hasattr(self._config, '_frozen') and self._config._frozen:
+            msg = "Cannot modify a frozen Config. "\
+                  f"Attempting to set item {attr} to value {value}"
+            raise FieldValidationError(self._field, self._config, msg)
+
+        if attr not in (self.__dict__.keys() | type(self).__dict__.keys()):
+            name = _joinNamePath(self._config._name, self._field.name, attr)
+            if at is None:
+                at = getCallStack()
+            if isinstance(value, ConfigurableAction):
+                valueInst = type(value)(__name=name, __at=at, __label=label, **value._storage)
+            else:
+                valueInst = value(__name=name, __at=at, __label=label)
+            self._attrs[attr] = valueInst
+        else:
+            super().__setattr__(attr, value)
+
+    def __getattr__(self, attr):
+        if attr in object.__getattribute__(self, '_attrs'):
+            return self._attrs[attr]
+        else:
+            super().__getattribute__(attr)
+
+    def __delattr__(self, name):
+        if name in self._attrs:
+            del self._attrs[name]
+        else:
+            super().__delattr__(name)
+
+    def __iter__(self) -> Generator[ConfigurableAction, None, None]:
+        yield from self._attrs.values()
```

Comment:
Most curiousity: why the extra generator indirection, rather than just `return iter(self._attrs.values())`?

Reply:
There is no compelling reason that either of these two methods do this. Its left over from some random early on idea that I moved on from and just never thought to change here. One note I will make on this, there is a subtle usage difference when it comes to `return iter` `yield from` `yield` when it come to what information is available in traceback messages, but in this case it really does not matter.